radixdlt/src/main/java/com/radixdlt/middleware2/store/CommittedAtomsStore.java

```diff
@@ -77,30 +84,42 @@ public CommittedAtomsStore(
 	) {
 		this.committedAtomSender = Objects.requireNonNull(committedAtomSender);
 		this.store = Objects.requireNonNull(store);
+		this.persistentVertexStore = Objects.requireNonNull(persistentVertexStore);
 		this.commandToBinaryConverter = Objects.requireNonNull(commandToBinaryConverter);
 		this.clientAtomToBinaryConverter = Objects.requireNonNull(clientAtomToBinaryConverter);
 		this.atomIndexer = Objects.requireNonNull(atomIndexer);
 		this.serialization = Objects.requireNonNull(serialization);
 		this.hasher = hasher;
 	}
 
-	private Optional<ClientAtom> getAtomByParticle(Particle particle, boolean isInput) {
+	private boolean getAtomByParticle(Particle particle, boolean isInput) {
 		final byte[] indexableBytes = EngineAtomIndices.toByteArray(
 		isInput ? EngineAtomIndices.IndexType.PARTICLE_DOWN : EngineAtomIndices.IndexType.PARTICLE_UP,
 			Particle.euidOf(particle, hasher)
 		);
-		SearchCursor cursor = store.search(StoreIndex.LedgerIndexType.UNIQUE, new StoreIndex(indexableBytes), LedgerSearchMode.EXACT);
-		if (cursor != null) {
-			return store.get(cursor.get())
-				.flatMap(ledgerEntry ->  {
-					// TODO: Remove serialization/deserialization
-					StoredCommittedCommand committedCommand = commandToBinaryConverter.toCommand(ledgerEntry.getContent());
-					ClientAtom clientAtom = committedCommand.getCommand().map(clientAtomToBinaryConverter::toAtom);
-					return Optional.of(clientAtom);
-				});
-		} else {
-			return Optional.empty();
-		}
+		return store.contains(this.transaction, StoreIndex.LedgerIndexType.UNIQUE, new StoreIndex(indexableBytes), LedgerSearchMode.EXACT);
+	}
+
+	@Override
+	public void startTransaction() {
+		this.transaction = store.createTransaction();
```

Comment:
Given that nested transactions are not supported here, this should throw if `this.transaction != null`.

Actually this implementation smells a bit, given that the transaction is hidden from API clients and just happens to be used.  Needs to be refactored so that transaction can be passed in to actual storage methods.